docker/cloudbuild.yaml

```diff
@@ -0,0 +1,22 @@
+steps:
+# Build the main getsentry/sentry image
+- name: 'gcr.io/cloud-builders/docker'
+  args: [ 'build', '--build-arg', 'SOURCE_COMMIT=$COMMIT_SHA', '-t', 'gcr.io/$PROJECT_ID/sentry:$COMMIT_SHA', '-t', 'gcr.io/$PROJECT_ID/sentry:$SHORT_SHA', '-t', 'gcr.io/$PROJECT_ID/sentry:latest', '-f', './docker/Dockerfile', '.' ]
+# Derive the -onbuild variant
+- name: 'gcr.io/cloud-builders/docker'
+  args: [ 'build', '-t', 'gcr.io/$PROJECT_ID/sentry:$COMMIT_SHA-onbuild', '-t', 'gcr.io/$PROJECT_ID/sentry:$SHORT_SHA-onbuild', '-t', 'gcr.io/$PROJECT_ID/sentry:latest-onbuild', '-f', './docker/onbuild/Dockerfile', '.' ]
+# Smoke tests
+- name: 'gcr.io/cloud-builders/docker'
+  args: ['config', 'get', 'system.secret-key']
+  env:
+    - SENTRY_SKIP_BACKEND_VALIDATION=1
+    - SENTRY_SECRET_KEY=abc
+    - SENTRY_REDIS_HOST=localhost
+timeout: 1200s
+images:
```

Comment:
I thought our plan was to still push to docker hub and not push into our GCR? We can use Cloud Builder to build, but then push to public Docker Hub.

I'd prefer we do this, but if there was some reason not to, there are some changes we need to make here.

* We always explicitly use `us.gcr.io` as our registry instead of the generic `gcr.io`.
* PROJECT_ID is potentially not going to be right, but we'd need to just discuss what project and where it's publishing to. It might be right, just not sure since we don't have the job setup yet afaik.

Reply:
The job is set up. My end goal is to still push to Docker Hub, just in a follow up PR as it requires a bot account, logging in to Docker Hub etc.